src/lib.rs

```diff
@@ -84,17 +157,70 @@ impl<K: std::hash::Hash + PartialEq, V> HashTable<K, V> {
         None
     }
 
+    pub fn get_mut(&mut self, k: &K) -> Option<&mut V> {
+        let hash = self.hasher.hash(k);
+        let bucket_index = hash as usize % self.buckets.len();
+        for (ek, v) in &mut self.buckets[bucket_index] {
+            if ek == k {
+                return Some(v);
+            }
+        }
+        None
+    }
+
     pub fn capacity(&self) -> usize {
         self.buckets.len()
     }
+
+    pub fn entry(&mut self, k: K) -> Entry<'_, K, V, H> {
+        if self.get_mut(&k).is_some() {
```

Comment:
It is very silly that I call get_mut twice: once here to check which entry type to return and then again when borrowing the value during a call to entry's or_insert. I tried borrowing the value mutably for occupied entries and for vacant entries, borrowing self. I thought that since they were divergent paths this should have been fine, but it wasn't and the compiler was unhappy:
```
pub enum Entry<'a, K, V, H>
where
    K: Hash,
    H: SimpleHasher<K>,
{
    Occupied {
        v: &'a mut V,
    },
    Vacant {
        ht: &'a mut HashTable<K, V, H>,
        k: K,
    },
}

error[E0499]: cannot borrow `*self` as mutable more than once at a time
   --> src/lib.rs:180:40
    |
175 |     pub fn entry(&mut self, k: K) -> Entry<'_, K, V, H> {
    |                  - let's call the lifetime of this reference `'1`
176 |         let v = self.get_mut(&k);
    |                 ---- first mutable borrow occurs here
177 |         if let Some(ov) = v {
178 |             return Entry::Occupied { v: ov };
    |                    ------------------------- returning this value requires that `*self` is borrowed for `'1`
179 |         } else {
180 |             return Entry::Vacant { ht: self, k };
    |                                        ^^^^ second mutable borrow occurs here
```
I think this is related to the non lexical lifetime issues since my exact case is provided here: https://github.com/rust-lang/rust/issues/21906#issuecomment-460963210
For now, we will just need to call get_mut twice.